pkg/remediator/watch/filteredwatcher.go

```diff
@@ -377,11 +378,16 @@ func (w *filteredWatcher) start(ctx context.Context, resourceVersion string) (bo
 	// We want to avoid situations of hanging watchers. Stop any watchers that
 	// do not receive any events within the timeout window.
 	timeoutSeconds := int64(minWatchTimeout.Seconds() * (rand.Float64() + 1.0))
+	// This label is added to all managed objects by the applier
+	selector := labels.SelectorFromSet(map[string]string{
+		metadata.ManagedByKey: metadata.ManagedByValue,
+	})
 	options := metav1.ListOptions{
 		AllowWatchBookmarks: true,
 		ResourceVersion:     resourceVersion,
 		TimeoutSeconds:      &timeoutSeconds,
 		Watch:               true,
+		LabelSelector:       selector.String(),
```

Comment:
I assume that we would receive a watch event if the object had the label when the watch started, but I need to confirm that assumption.

Yes this would be new behavior. We do tell users that this label is [not safe to modify](https://cloud.google.com/anthos-config-management/docs/reference/labels-and-annotations), although I'm not sure whether that is sufficient.

@janetkuo WDYT?

Reply:
It's somewhat breaking even though we tell users not to touch the label. The question is how big of a deal it is. The label would be re-applied when that resource is applied by the reconciler next time, right?

Reply:
Yes that's a good point, I believe you are correct that the reconciler would re-apply the label the next time the sync loop runs.

Reply:
Could this be used as a mechanism to maliciously cause temporary drift? Just remove the label and you buy yourself an hour before it's re-applied? 

I like the optimization I'm just not entirely convinced it doesn't neuter the remediator's reason for existing. 

Should the remediator be able to prevent hostile drift or is it really only for eventually reverting accidental drift?

Reply:
The remediate can receive the watch event after the watch is started. What if a separate process deletes/updates the managedBy annotation after the object is applied, AND before the watch is set up?

Reply:
I tested this locally by adding a sleep after the applier runs and before the watches are started, to make it easier to reproduce. The behavior in this race condition is that the remediator will not receive watch events for the object if the label is removed before the watch starts.

First, I believe this race condition is applicable even without the label selector - i.e. any changes made after the object is applied but before the watch is started will be ignored by the remediator. The primary difference with the label selector is that there could be additional drift after the watch is started (assuming the label is removed before the watch starts).

Otherwise the label would get reintroduced the next time the applier runs, at which point the remediator would receive watch events for the object (unless the race condition happens again).

If we view this as a problem, a few mitigations come to mind:
- We could start the remediator watches before the applier runs. Perhaps this could be done after the declared resource cache is updated, or we could enqueue events but not act on them until the applier finishes running.
- We could have the remediator perform a GET on all objects in the declared cache after the watches start to ensure no drift has occurred since the applier ran.

I'm not sure whether this particular race condition is a large enough concern to introduce such a mitigation.

Reply:
IMO the simplest solution would be the first one I mentioned. We can start the remediator watches after the declared resource cache is updated but before the applier runs. This way the remediator should do a no-op for the updates done by the applier, but also correct for drift introduced by any separate processes.

Reply:
I don't think the migration should block this PR given that it is not a new issue introduced by the label selector.

/lgtm

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1070